src/System.Private.ServiceModel/tools/PRService/pr.ashx

```diff
@@ -60,26 +62,35 @@ public class PullRequestHandler : IHttpHandler
             context.Response.Write(HttpUtility.HtmlEncode(result));
             return;
         }
-        else
+
+        success = false;
+
+        // Check inputs to the script. PR has precedence
+        uint pr;
+        if (!string.IsNullOrEmpty(prString) && uint.TryParse(prString, out pr))
         {
-            // Check inputs to the script
-            uint pr;
-            if (!string.IsNullOrEmpty(prString) && uint.TryParse(prString, out pr))
-            {
-                success = SyncToPr(pr, result);
-            }
-            else
+            success = SyncToPr(pr, result);
+        }
+        else if (!string.IsNullOrEmpty(branchString))
```

Comment:
If you do make that assumption, the whole `if (!string.IsNullOrEmpty(branchString))` can be saved (and probably many other things like the validation of "master") :smile:. Maybe I did not use a good example, this code will work for `pr.ashx?pr=bogus&branch=master`.

Validating input parameters at beginning (you decide where is the right place) is a good practice many people like to do and it helps to make code after that simpler. I am just trying to see how we can make this code reliable as we both know we want to avoid updating this code once it's in action, but I will not block you from merging if you don't think so.